src/formatDate.js

```diff
@@ -7,8 +7,54 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dayArr = date.split(fromFormat[3]);
+  let [year, month, day] = dayArr;
+  const newFormat = [];
+
+  if (fromFormat.indexOf('YYYY') === 0 || fromFormat.indexOf('YY') === 0) {
+    [year, month, day] = dayArr;
+  } else if (
+    fromFormat.indexOf('YYYY') === 1 ||
+    fromFormat.indexOf('YY') === 1
+  ) {
+    [month, year, day] = dayArr;
+  } else if (fromFormat.indexOf('MM') === 0) {
+    [month, day, year] = dayArr;
+  } else {
+    [day, month, year] = dayArr;
+  }
```

Comment:
The conditional block for reassigning `year`, `month`, and `day` is redundant since you've already destructured them on line 13. You can rely on the order of `fromFormat` directly when destructuring.